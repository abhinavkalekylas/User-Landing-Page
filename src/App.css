*,
*::after,
*::before {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

body {
  position: relative;
  background-color: rgb(104, 104, 104);
}

nav {
  border-radius: 0.625rem;
}
