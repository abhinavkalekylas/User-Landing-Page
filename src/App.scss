$body-color: rgb(104, 104, 104);

*,
*::after,
*::before {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

body {
  position: relative;
  background-color: $body-color;
}


