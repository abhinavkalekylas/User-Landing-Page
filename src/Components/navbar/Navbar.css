nav {
    width: 100%;
}

.navbar {
    background: black;
    justify-content: space-between;
    margin-top: 0.25rem;
}

.navbar-brand {
    margin-left: 0.75rem;
    margin-right: 0.75rem;
}

form {
    display: flex;
    flex-direction: row;
}

.form-select {
    margin-top: 0.3em;
    margin-bottom: 0.3em;
}

.adduser > button {
    width: 120px;
    height: 45px;
    margin-left: 1rem;
    margin-right: 1rem;
    margin-top: 0.5rem;
    margin-bottom: 0.5rem;
}