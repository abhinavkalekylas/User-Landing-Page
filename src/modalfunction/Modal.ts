export interface User {
  id: number;
  name: string;
  email: string;
  gender: GenderType;
  status: StatusType;
}

export enum GenderType {
  MALE = "male",
  FEMALE = "female",
}

export enum StatusType {
  ACTIVE = "active",
  INACTIVE = "inactive",
}

export const validateEmail = (email: string) => {
  const regex = /^[^\s@]+@[^\s@]+\.[^\s@]{2,}$/i;
  if (regex.test(email)) {
    return true;
  }
  return false;
};
