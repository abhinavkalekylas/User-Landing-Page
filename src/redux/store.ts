import { legacy_createStore, compose } from "redux";
import reducers from "./reducer/todo"

// declare global {
//     interface Window {
//       __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
//     }
//   }

// const store = legacy_createStore(reducers, window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose);

const store = legacy_createStore(reducers, {});

export default store;