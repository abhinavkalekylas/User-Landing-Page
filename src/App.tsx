import "bootstrap/dist/css/bootstrap.min.css";
import "./App.css";
import { BrowserRouter as Router } from "react-router-dom";
import Navbar from "./Components/navbar/Navbar";
import UserList from "./Components/userList/userList";

const App = () => {
  return (
    <Router>
      <div className="container">
        <Navbar />
        <UserList />
      </div>
    </Router>
  );
};

export default App;
