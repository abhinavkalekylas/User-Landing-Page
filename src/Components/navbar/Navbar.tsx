import React, { useEffect, useState } from "react";
import { Button } from "react-bootstrap";
import "./Navbar.scss";
import Modaldata from "../modal/Modaldata";
import { GenderType, StatusType } from "../../modalfunction/Modal";
import { useDispatch } from "react-redux";
import { filterUser } from "../../redux/action/todo";

const Navbar = () => {
  // user state to show/hide modal
  const [addUserModal, setAddUserModal] = useState(false);

  const closeModal = () => setAddUserModal(false);
  const openModal = () => setAddUserModal(true);

  const dispatch = useDispatch();
  const [filter, setFilter] = useState({
    selectedGender: "",
    selectedStatus: "",
  });

  const handleChangeStatus = (e: {
    target: { name: string; value: string };
  }) => {
    setFilter((prevType) => ({
      ...prevType,
      selectedStatus: e.target.value,
    }));
  };

  const handleChangeGender = (e: {
    target: { name: string; value: string };
  }) => {
    setFilter((prevType) => ({
      ...prevType,
      selectedGender: e.target.value,
    }));
  };

  useEffect(() => {
    dispatch(filterUser(filter));
  }, [filter]);

  return (
    <nav className="navbar navbar-dark">
      <a href="/" className="navbar-brand ">
        User Landing Page
      </a>

      <form className="nav_form form-inline">
        <span className="filter_title">Filter by:</span>
        <select
          className="filter form-select form-select-lg gender_select"
          aria-label=".form-select-lg example"
          onChange={handleChangeGender}
        >
          <option value="">None</option>
          <option value={GenderType.MALE}>Male</option>
          <option value={GenderType.FEMALE}>Female</option>
        </select>
        <select
          className="filter form-select form-select-lg status_select"
          aria-label=".form-select-lg example"
          onChange={handleChangeStatus}
        >
          <option value="">None</option>
          <option value={StatusType.ACTIVE}>Active</option>
          <option value={StatusType.INACTIVE}>Inactive</option>
        </select>
        <div className="adduser">
          <Button
            color="success"
            className="btn_add btn btn-success btn-outline-light"
            onClick={openModal}
          >
            Add User
          </Button>
        </div>
      </form>
      <Modaldata addUserModal={addUserModal} closeModal={closeModal} />
    </nav>
  );
};

export default Navbar;
