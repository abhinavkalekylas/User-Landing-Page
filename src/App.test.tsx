import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import { token } from "./modalfunction/Modal";
import App from "./App";
import Navbar from "./Components/navbar/Navbar";
import UserList from "./Components/userList/userList";
import { Provider } from "react-redux";
import store from "./redux/store";

it("renders correctly App component", () => {
  const wrapper = shallow(<App />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
