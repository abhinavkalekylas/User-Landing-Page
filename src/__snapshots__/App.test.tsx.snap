// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly App component 1`] = `
<App>
  <BrowserRouter>
    <Router
      location={
        Object {
          "hash": "",
          "key": "default",
          "pathname": "/",
          "search": "",
          "state": null,
        }
      }
      navigationType="POP"
      navigator={
        Object {
          "action": "POP",
          "back": [Function],
          "block": [Function],
          "createHref": [Function],
          "forward": [Function],
          "go": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "default",
            "pathname": "/",
            "search": "",
            "state": null,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <div
        className="container"
      >
        <Navbar>
          <nav
            className="navbar navbar-dark"
          >
            <a
              className="navbar-brand "
              href="/"
            >
              User Landing Page
            </a>
            <form
              className="form-inline"
            >
              <select
                aria-label=".form-select-lg example"
                className="form-select form-select-lg"
              >
                <option
                  value="none"
                >
                  None
                </option>
                <option
                  value="active"
                >
                  Active
                </option>
                <option
                  value="inactive"
                >
                  Inactive
                </option>
                <option
                  value="male"
                >
                  Male
                </option>
                <option
                  value="female"
                >
                  Female
                </option>
              </select>
              <div
                className="adduser"
              >
                <button
                  className="btn btn-success btn-outline-light"
                  color="success"
                >
                  Add User
                </button>
              </div>
            </form>
          </nav>
        </Navbar>
        <UserList>
          <div
            className="container"
          >
            <Loader>
              <div
                className="container loader"
              >
                <h2>
                  Loading...  Please wait..
                </h2>
                <div
                  aria-hidden="true"
                  className="spinner-border ms-auto"
                  role="status"
                />
              </div>
            </Loader>
          </div>
        </UserList>
      </div>
    </Router>
  </BrowserRouter>
</App>
`;
